lib/match-across-boundaries.js

```diff
@@ -0,0 +1,42 @@
+'use strict';
+
+var mutuallyExclusiveDependencyKeys = {
+  dependencies: [
+    'devDependencies',
+    'optionalDependencies'
+  ],
+  devDependencies: [
+    'dependencies'
+  ],
+  optionalDependencies: [
+    'dependencies'
+  ]
+};
+
+// user has dependency dep@1
+// remote updated devDependency dep@1 -> dep@2
+// result: dependency is updated dep@1 -> dep@2
+module.exports = function matchAcrossBoundaries(merger) {
+  Object.keys(mutuallyExclusiveDependencyKeys).forEach(function(dependencyKey) {
+    var sourceDeps = merger.source[dependencyKey];
+    var ourDeps = merger.ours[dependencyKey];
+    var thierDeps = merger.theirs[dependencyKey];
```

Comment:
not a big deal, but this is a typo "thier" ->"their"

Reply:
there there, it'll be ok... 😂

Reply:
fixed